src/BusinessEvents.SubscriptionEngine.Handlers/Handler.cs

```diff
@@ -59,189 +46,30 @@ public Handler(IContainer container)
+            await eventStreamProcessor.Process(kinesisEvent);
         }
 
-        public async Task NotifySubscriber(LambdaInvocationPayload lambdaInvocationPayload, ILambdaContext context)
+        private static string GetAccountId(string functionArn)
         {
-            var watch = System.Diagnostics.Stopwatch.StartNew();
-            var logger = context.Logger;
-
-            logger.Log($"Lambda Event: {JsonConvert.SerializeObject(lambdaInvocationPayload)}");
-
-            Event @event = null;
-
-            try
-            {
-                @event = JsonConvert.DeserializeObject<Event>(lambdaInvocationPayload.EncryptedEvent.ToUncompressedString().Decrypt());
-                var serviceProcess = Container.Resolve<IServiceProcess>();
-                await serviceProcess.NotifySubscriber(lambdaInvocationPayload.Subscription, @event);
-            }
-            catch (Exception e)
-            {
-
-                await MarkAsDeadLetter(@event, context.FunctionName, e);
-                logger.LogLine($"Error: {e}");
-            }
-
-            watch.Stop();
-            logger.Log($"Events Processed  {lambdaInvocationPayload.Subscription.Type} MessageId: {@event?.Message?.Header?.MessageId} Time taken: {(watch.ElapsedMilliseconds/1000)} secs");
+            var arnItems = functionArn.Split(':');
+            return arnItems[4];
         }
 
-        public async Task HandleDeadLetterQueue(object cloudwatchEvent, ILambdaContext context)
+        public async Task ProcessDynamoDbStream(DynamoDBEvent dynamoDbEvent, ILambdaContext context)
         {
-            try
-            {
-                var dlq = Container.ResolveKeyed<IQueue>("DLQ");
-                List<Message> messages;
-                do
-                {
-                    messages = await dlq.GetMessages();
-                    Console.WriteLine($"Processing {messages.Count} messages");
-                    foreach (var message in messages)
-                    {
-                        Console.WriteLine($"MessageId: {message.MessageId}, ReceiptHandle: {message.ReceiptHandle} Dead Letter Message: {message.Body}");
-                        var success = await dlq.DeleteMessage(message.ReceiptHandle);
-                        if (!success)
-                        {
-                            Console.WriteLine($"Unable to Delete Message From DLQ: ReceipeHandle: {message.ReceiptHandle}");
-                        }
-                    }
-                }
-                while (messages.Count > 0 && context.RemainingTime.TotalMilliseconds > 2000);
-            }
-            catch (Exception e)
-            {
-                Console.WriteLine($"Error: Processing DLQ Error: {JsonConvert.SerializeObject(e)}");
-            }
+            var dataStreamProcessor = Container.Resolve<IDataStreamProcessor>();
+            await dataStreamProcessor.Process(dynamoDbEvent, GetAccountId(context.InvokedFunctionArn));
```

Comment:
We can set ACCOUNT_ID here, so we don't have to pass in the account id after calling GetAccountId(context.InvokedFunctionArn)
Environment.SetEnvironmentVariable("ACCOUNT_ID", awsAccountId);